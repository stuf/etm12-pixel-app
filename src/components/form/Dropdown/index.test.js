import * as React from 'karet';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import * as K from 'kefir';

import Index from './index';

it('matches snapshot', () => {
  const xs = [{ name: 'top' }, { name: 'kek' }];
  const xs$ = K.constant(xs);

  const w1 = mount(<Index />);
  const w2 = mount(<Index items={xs$} />);
  const w3 = mount(<Index items={xs} />);

  expect(toJson(w1)).toMatchSnapshot('empty dropdown');
  expect(toJson(w2)).toMatchSnapshot('non-empty dropdown');
  expect(toJson(w3)).toMatchSnapshot('non-empty dropdown (plain)');
});
