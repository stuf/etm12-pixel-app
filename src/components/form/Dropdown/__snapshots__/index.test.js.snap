// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot: empty dropdown 1`] = `
<Dropdown>
  <div
    className="root"
  >
    <select
      onChange={[Function]}
    />
  </div>
</Dropdown>
`;

exports[`matches snapshot: non-empty dropdown (plain) 1`] = `
<Dropdown
  items={
    Array [
      Object {
        "name": "top",
      },
      Object {
        "name": "kek",
      },
    ]
  }
>
  <div
    className="root"
  />
</Dropdown>
`;

exports[`matches snapshot: non-empty dropdown 1`] = `
<Dropdown
  items={
    P {
      "_currentEvent": Object {
        "current": true,
        "type": "value",
        "value": Array [
          Object {
            "name": "top",
          },
          Object {
            "name": "kek",
          },
        ],
      },
    }
  }
>
  <FromClass
    args={
      Array [
        "div",
        Object {
          "__self": undefined,
          "__source": Object {
            "fileName": "/home/stuf/Projects/Lab/etm12-pixel-app/src/components/form/Dropdown/index.js",
            "lineNumber": 17,
          },
          "className": "root",
        },
        AnonymousObservable {
          "_$handleAny": null,
          "_activating": false,
          "_active": false,
          "_alive": false,
          "_currentEvent": Object {
            "type": "value",
            "value": undefined,
          },
          "_dispatcher": null,
          "_getInitialCurrent": null,
          "_logHandlers": null,
          "_name": "constant.flatMapLatest.toProperty",
          "_source": null,
          "_spyHandlers": null,
        },
      ]
    }
  >
    <div
      className="root"
    />
  </FromClass>
</Dropdown>
`;
